cmd/gentypes/gentypes.go

```diff
@@ -124,10 +124,22 @@ func parsePropertyType(propValue map[string]interface{}) string {
 		}
 
 	case []interface{}:
+		// This field is polymorphic so it needs a generic type.
+		for _, p := range propType.([]interface{}) {
```

Comment:
alternatively, I think, at line 84, you can do

```
switch typ := propType.(type) {
  case string: 
     ...
  case []interface{}:
     if _, p := range typ {
        ...
     }
     
```

Reply:
Good idea! I got confused as in the previous case the string type was correctly inferred but not for `[]interface{}` here. Anyway, `typ` is much nicer.